### PR TITLE
Add Application Gateway to SF deploy code

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/Microsoft.DncEng.DeployServiceFabricCluster/GatewaySettings.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        public Dictionary<string, string>? NeededSecurityGroupRules
+        {
+            get
+            {
+                if (ExternalPorts == null)
+                {
+                    return null;
+                }
+                return new[]
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.DeployServiceFabricCluster/GatewaySettings.cs:16]: This... probably need to get broken out a bit. :-) It's a LOT of link and type bouncing back and forth.

I think this is just a new dictionary, and then add the various ports in a loop?

File: fix-npm-build-for-vs2019.cmd [hunk 2]
```diff
@@ -0,0 +1,2 @@
+npm config set msbuild_path "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe" -g
```
Comment (ChadNedzlek) [inline fix-npm-build-for-vs2019.cmd:1]: This file scares me a little. :-)

File: src/Microsoft.DncEng.DeployServiceFabricCluster/GatewaySettings.cs (hunk 1 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.DeployServiceFabricCluster/GatewaySettings.cs:16]: Yea, i'll change it.

File: fix-npm-build-for-vs2019.cmd (hunk 2 above)
Comment (alexperovich) [reply fix-npm-build-for-vs2019.cmd:1]: node-sass forces an old version of node-gyp, which is broken if your machine only has vs 2019 on it.

**Pull request merged**